### DISCUSSION TITLE
Relation between `MeasurementResult` and the circuit conditions that originated it

### DISCUSSION THREAD

**Discussion by EmilianoG-byte** [unitaryfund/mitiq#2043] [Q&A] [answered]
Hi!

In the `Execute` class the method `_call_executor` stores the results and the circuits run in the variables `_quantum_results` and `_executed_circuits`, respectively. However, this does not provide information of what "conditions" was each circuit run under, right? For instance, what method was used to amplify the noise, what noise scale, etc. 

Is this correct? Or am I missing that this information can be accessed in a different way?

**Comment by andreamari** [comment, marked as answer]
Hi @EmilianoG-byte, 
correct. There is no log of mitigation methods or similar hyper-parameters in `Executor` objects or in `MeasurementResult` objects.

Hint: If you are using ZNE, you can deduce the noise scale factor associated to a specific circuit $C_j$  in `_executed_circuits` by evaluating the number $n_j$ of gates in $C_j$ and comparing it to the number of gates of the unmitigated circuit $n_{unmitigated}$. The scale factor is approximately given by $\lambda_j = n_j / n_{unmitigated}$.

Note also that, by default, Mitiq uses `scale_factors = [1, 2, 3]` and `fold_gates_at_random` as a noise scaling method.

**Reply by EmilianoG-byte** [reply]
I see, thank you!
